server/controllers/receipt-controller.js

```diff
@@ -32,16 +32,27 @@ const createReceipt = async (req, res, next) => {
 // @access Private
 const updateReceipt = async (req, res, next) => {
     try {
+        // check
+        const receipt = await Receipt.findById(req.params.id);
+        if (!receipt) {
+            const error = new HttpError('Receipt not found', 404);
+            return next(error);
+        }
+        if (receipt.user != req.userData.userId) {
+            const error = new HttpError('Not Authorized', 401);
+            return next(error);
+        }
+
         //update receipt
-        const receipt = await Receipt.findOneAndUpdate(
+        const newReceipt = await Receipt.findOneAndUpdate(
             { _id: req.params.id },
             req.body,
             { new: true }
         );
-        await receipt.save();
-        return res.json(receipt);
+        await newReceipt.save();
+        return res.json(newReceipt);
```

Comment:
I didn't get this logic?

Reply:
findOneAndUpdate's first param finds a receipt that has the id of the receipt id we are looking for, second param is to change the fields that are passed in from the req.body, third param makes sure that the newly updated object is returned

Reply:
ok, it's good but I will try to improve it more, for now you go with it. good job